starter-code/routes/auth.js

```diff
@@ -0,0 +1,119 @@
+'use strict';
+
+const express = require('express');
+const router = express.Router();
+
+// User model
+const User = require('../models/user').User;
+
+// BCrypt
+const bcrypt = require('bcrypt');
+const bcryptSalt = 10;
+
+// --- Sign up --- //
+router.get('/signup', function(req, res, next) {
+    res.render('auth/signup');
+});
+
+// --- Redirect --- //
+router.get('/registered', function(req, res, next) {
+    res.render('auth/registered');
+});
+
+router.post('/signup', function(req, res, next) {
+    const username = req.body.username;
+    const password = req.body.password;
+    const salt = bcrypt.genSaltSync(bcryptSalt);
```

Comment:
Better if you do the encryption and encapsulate the user (lines 26 to 32) after checking the username is not taken.